### ISSUE TITLE
RELATIVE_URL_ROOT does not work well

### ISSUE THREAD

**Issue by nicolasgoudard** [sameersbn/docker-gitlab#2600]
Gitlab does not follow the subpath defined in the RELATIVE_URL_ROOT environment variable. I saw that gitlab-ce recommends using a subdomain instead, but that's not possible in my infrastructure. So can't I use Gitlab ?

Here the installation script : 

```
docker run --name gitlab-postgresql -d \
    --env 'DB_NAME=gitlabhq_production' \
    --env 'DB_USER=gitlab' --env 'DB_PASS=password' \
    --env 'DB_EXTENSION=pg_trgm,btree_gist' \
    --volume /Users/nicolas/docker/gitlab/postgresql:/var/lib/postgresql \
    sameersbn/postgresql:12-20200524


docker run --name gitlab-redis -d \
    --volume /Users/nicolas/docker/gitlab/redis:/data \
    redis:6.2

docker run --name gitlab -d \
    --link gitlab-postgresql:postgresql --link gitlab-redis:redisio \
    --publish 10022:22 --publish 8083:80 \
    --env 'GITLAB_PORT=8083' --env 'GITLAB_SSH_PORT=10022' \
    --env 'GITLAB_SECRETS_DB_KEY_BASE=long-and-random-alpha-numeric-string' \
    --env 'GITLAB_SECRETS_SECRET_KEY_BASE=long-and-random-alpha-numeric-string' \
    --env 'GITLAB_SECRETS_OTP_KEY_BASE=long-and-random-alpha-numeric-string' \
    -**-env 'RELATIVE_URL_ROOT=/gitlab'**  \
    --volume /Users/nicolas/docker/gitlab/gitlab:/home/git/data \
    sameersbn/gitlab:15.0.3
```

As you can see it does not work :  It does not redirect to http://localhost:8083/gitlab/users/sign_in

```
$ curl http://localhost:8083/gitlab

<html><body>You are being <a href="**http://localhost:8083/users/sign_in**">redirected</a>.</body></html>                           
```

**Comment by kkimurak** [comment]
Yes `RELATIVE_URL_ROOT` is not supported for `sameersbn/gitlab`. Use [`GITLAB_RELATIVE_URL_ROOT`](https://github.com/sameersbn/docker-gitlab#gitlab_relative_url_root) instead.

**Comment by nicolasgoudard** [comment]
@kkimurak . Ok tanks it was a mistake but it still  does not work 
```
docker run --name gitlab-postgresql -d \
    --env 'DB_NAME=gitlabhq_production' \
    --env 'DB_USER=gitlab' --env 'DB_PASS=password' \
    --env 'DB_EXTENSION=pg_trgm,btree_gist' \
    --volume /Users/nicolas/docker/gitlab/postgresql:/var/lib/postgresql \
    sameersbn/postgresql:12-20200524

docker run --name gitlab-redis -d \
    --volume /Users/nicolas/docker/gitlab/redis:/data \
    redis:6.2

docker run --name gitlab -d \
    --link gitlab-postgresql:postgresql --link gitlab-redis:redisio \
    --publish 10022:22 --publish 8083:80 \
    --env 'GITLAB_PORT=8083' --env 'GITLAB_SSH_PORT=10022' \
    --env 'GITLAB_SECRETS_DB_KEY_BASE=long-and-random-alpha-numeric-string' \
    --env 'GITLAB_SECRETS_SECRET_KEY_BASE=long-and-random-alpha-numeric-string' \
    --env 'GITLAB_SECRETS_OTP_KEY_BASE=long-and-random-alpha-numeric-string' \
    --env 'GITLAB_RELATIVE_URL_ROOT=/gitlab'  \
    --volume /Users/nicolas/docker/gitlab/gitlab:/home/git/data \
    sameersbn/gitlab:15.0.3
```

It seems to redirect 
$ curl http://localhost:8083/gitlab
<html><body>You are being <a href="http://localhost:8083/gitlab/users/sign_in">redirected</a>.</body></html>

but when I use the browser it redirects to
http://localhost:8083/gitlab/users/password/edit?reset_password_token=8yJUv3QYzSdNCeBChUUt

with an error : 
500
Whoops, something went wrong on our end.
Request ID: 01G6WGQ0PC2RRK9DEJTDBG6DNC
Try refreshing the page, or going back and attempting the action again.
Please contact your GitLab administrator if this problem persists.

<img width="1017" alt="image" src="https://user-images.githubusercontent.com/6792673/176867156-4c5bddf6-30f3-4158-8dc0-ff2a4e574f32.png">

Please help me !

**Comment by nicolasgoudard** [comment]
> It takes few minutes to gitlab to launch up when you enabling relative url. In my environment, typically 20 min or something.
> Keep checking the logs by `docker logs --follow gitlab`. The message `Clearing cache` will appear when the compile done.

I have this message too

```
2022-07-01 09:13:10.716 UTC [141] WARNING:  could not open statistics file "pg_stat_tmp/global.stat": Operation not permitted
2022-07-01 09:14:11.091 UTC [191] ERROR:  duplicate key value violates unique constraint "user_details_pkey"
2022-07-01 09:14:11.091 UTC [191] DETAIL:  Key (user_id)=(1) already exists.
2022-07-01 09:14:11.091 UTC [191] STATEMENT:  /*application:web,correlation_id:01G6WGKXJ09AMDBTKVZVHA5B0S,db_config_name:main*/ INSERT INTO "user_details" ("user_id") VALUES (1) RETURNING "user_id"
2022-07-01 09:20:10.862 UTC [141] WARNING:  could not open statistics file "pg_stat_tmp/global.stat": Operation not permitted
2022-07-01 09:23:41.234 UTC [242] WARNING:  could not open statistics file "pg_stat_tmp/global.stat": Operation not permitted
2022-07-01 09:26:41.852 UTC [248] WARNING:  could not open statistics file "pg_stat_tmp/global.stat": Operation not permitted
2022-07-01 09:27:10.971 UTC [141] WARNING:  could not open statistics file "pg_stat_tmp/global.stat": Operation not permitted
```

**Comment by nicolasgoudard** [comment]
> It takes few minutes to gitlab to launch up when you enabling relative url. In my environment, typically 20 min or something.
> Keep checking the logs by `docker logs --follow gitlab`. The message `Clearing cache` will appear when the compile done.

tanks for your time and your answers 
did you see my message about postgresql logs ?

The compilation is very slow... The logs does not progress however my system is not full 

<img width="890" alt="image" src="https://user-images.githubusercontent.com/6792673/176869708-72b0025f-287a-4f6c-976a-936084f3a7d7.png">

**Comment by nicolasgoudard** [comment]
Ok  you said true but it works now !! Only wait and suddenly it works
Now I want to to migrate  my Synology gitlab 11.11.8  backup datas  in this new  installation. I don't know if it is possible . It is only simple repositories

-rwxr-xr-x    1 nicolas  staff        147 25 jui 00:01 artifacts.tar.gz
-rwxr-xr-x    1 nicolas  staff        183 25 jui 00:01 backup_information.yml
-rwxr-xr-x    1 nicolas  staff        147 25 jui 00:01 builds.tar.gz
drwxr-xr-x    3 nicolas  staff         96 25 jui 00:01 db
-rwxr-xr-x    1 nicolas  staff        146 25 jui 00:01 lfs.tar.gz
-rwxr-xr-x    1 nicolas  staff        146 25 jui 00:01 pages.tar.gz
drwx------    5 nicolas  staff        160 25 jui 00:01 repositories
-rwxr-xr-x    1 nicolas  staff     167674 25 jui 00:01 uploads.tar.gz

**Comment by kkimurak** [comment]
The backup must be a single tar archive named `xxxxxxxxxxx_yyyy_mm_dd_vv.vv.vv_gitlab_backup.tar`. `xxxx..` is number (as I remember unix time of snapshot or something). yyyy_mm_dd is backup date and vv.vv.vv is version of gitlab that backup is created. So you can 

Anyway, You must start from same version of gitlab when you created the backup. In this case it is `11.11.8`.  
Then, follow the [upgrade path instruction](https://docs.gitlab.com/ee/update/#version-specific-upgrading-instructions). Note that some background db migrations are introduced so you need to run a specific version for a while.

**Comment by nicolasgoudard** [comment]
> The backup must be a single tar archive named `xxxxxxxxxxx_yyyy_mm_dd_vv.vv.vv_gitlab_backup.tar`. `xxxx..` is number (as I remember unix time of snapshot or something). yyyy_mm_dd is backup date and vv.vv.vv is version of gitlab that backup is created. So try put them into a single tar archive and restore it in new installation.
> 
> Anyway, You must start from same version of gitlab when you created the backup. In this case it is `11.11.8`. Then, follow the [upgrade path instruction](https://docs.gitlab.com/ee/update/#version-specific-upgrading-instructions). Note that some background db migrations are introduced so you need to run a specific version for a while.

Thanks. But on docker hub the last  is 11.11.2 then 12.0  ( https://hub.docker.com/r/sameersbn/gitlab/tags?page=7 ) do you think it can work if I restore a 11.11.8 to 11.11.2 ?

I read the upgrade instructions but it is very difficult to move from 11.11.8 to 15.0.3

**Comment by frenchbeard** [comment]
Considering the date of the last comment, this may come a little late, but regarding the 11.11.8 data in 11.11.2, I had to do the same on 12.3.X instances, and it worked without any issue (still had to reupload users' and deploy SSH keys, but besides that, no issues), and then, as @kkimurak suggested, even though a little tedious, the upgrade path can be followed.
An idea to make it a little quicker : do not define the GITLAB_RELATIVE_URL_PATH until you upgrade to the last version of your upgrade, it would prevent having to fully recompile the assets for each
